kkanbu/board/helpers/filters.py

```diff
@@ -0,0 +1,68 @@
+from django.db.models import Q
+from django.utils.encoding import force_str
+from django.utils.translation import gettext_lazy as _
+from rest_framework import filters
+from rest_framework.compat import coreapi, coreschema
+
+
+class TagFilter(filters.BaseFilterBackend):
+    tag_param = "tags"
+    tag_title = _("Tag")
+    tag_description = _("The tags match.")
+
+    def get_tag_terms(self, request):
+        """
+        Search terms are set by a ?tags=... query parameter,
+        and may be comma and/or whitespace delimited.
+        """
+        params = request.query_params.get(self.tag_param, "")
+        params = params.replace("\x00", "")  # strip null characters
+        params = params.replace(",", " ")
+        return params.split()
+
+    def filter_queryset(self, request, queryset, view):
+        tag_terms = self.get_tag_terms(request)
+
+        if not tag_terms:
+            return queryset
+
+        queries = [Q(tags__name=tag_term) for tag_term in tag_terms]
+        queryset_list = [queryset.filter(query) for query in queries]
+        for q in queryset_list:
+            queryset = queryset.intersection(q)
+        return queryset
+
+    def get_schema_fields(self, view):
+        return [
+            coreapi.Field(
+                name=self.tag_param,
+                required=False,
+                location="query",
+                schema=coreschema.String(
+                    title=force_str(self.tag_title),
+                    description=force_str(self.tag_description),
+                ),
+            )
+        ]
+
+    def get_schema_operation_parameters(self, view):
+        return [
+            {
+                "name": self.tag_param,
+                "required": False,
+                "in": "query",
+                "description": force_str(self.tag_description),
+                "schema": {
+                    "type": "string",
+                },
+            },
+        ]
+
+
+class PostOrderingFilter(filters.OrderingFilter):
+    # The URL query parameter used for the ordering.
+    ordering_param = "ordering"
```

Comment:
코드 살펴보면 이미 default로 ordering이 설정되어 있어서 불필요한 코드로 보임!    

# Filtering
    'SEARCH_PARAM': 'search',
    'ORDERING_PARAM': 'ordering',

Reply:
공식문서 : By default, the query parameter is named 'ordering', but this may by overridden with the ORDERING_PARAM setting. / 링크 : https://www.django-rest-framework.org/api-guide/filtering/#orderingfilter

Reply:
아 엉 맞는데 혹시나 query param을 바꿀수도 있고, 프론트엔드나 제3자가 코드볼 때 한번에 알아볼 수 있게 하려고 이건 일부로 넣어놨어~